Assignment3/HaveWeMet/Tests/LocationHistoryAnalyzerTests.cs

```diff
@@ -0,0 +1,160 @@
+﻿using HaveWeMet;
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Tests
+{
+    class LocationHistoryAnalyzerTests
+    {
+        [Test]
+        public static void CheckAlibiReturnsLocationsForDateTest()
+        {
+            var locations = new List<Location> { new Location(){ timestampMs = "1484242416215",
+                                                                 latitudeE7 =  436615711,
+                                                                 longitudeE7 = -702530221},
+                                                new Location(){ timestampMs = "1484242459563",
+                                                                latitudeE7 = 436615711,
+                                                                longitudeE7 = -702530221},
+                                                new Location(){timestampMs = "1554074042394",
+                                                               latitudeE7 = 436611851,
+                                                               longitudeE7 = -702512553},
+                                                 new Location(){timestampMs = "1554074042999",
+                                                               latitudeE7 = 436611851,
+                                                               longitudeE7 = -702512553}
+                                                };
+            var results = LocationHistoryAnalyzer.CheckAlibi(new DateTimeOffset(new DateTime(2017, 1, 12)), locations);
+            Assert.AreEqual(2, results.Count);
+            CollectionAssert.Contains(results, locations[0]);
+            CollectionAssert.Contains(results, locations[1]);
+            CollectionAssert.DoesNotContain(results, locations[2]);
```

Comment:
locations[3] seems lonely 
also comments might be helpful for what the timestampMs is equal to (only when it's necessary like when your comparing with dates, not really needed for the test where you just deserialize it).
```
new Location(){ timestampMs = "1484242416215",     // Thursday, January 12, 2017 12:33:36.215 PM
```
But overall there's a lot of tests that cover lots of cases and they're very clear in what they should be testing for! Good job not only on these but the whole project!

Reply:
Thank you for the suggestion, that's a good idea. I will do that next time!